partiql-parser/src/main/kotlin/org/partiql/parser/internal/V1PartiQLParserDefault.kt

```diff
@@ -0,0 +1,2195 @@
+        }
+
+        override fun visitEdgeSpecUndirectedLeft(ctx: GeneratedParser.EdgeSpecUndirectedLeftContext): AstNode {
+            val edge = visitEdgeSpec(ctx.edgeSpec())
+            return edge.copy(direction = GraphDirection.LEFT_OR_UNDIRECTED())
+        }
+
+        override fun visitEdgeSpecUndirectedRight(ctx: GeneratedParser.EdgeSpecUndirectedRightContext): AstNode {
+            val edge = visitEdgeSpec(ctx.edgeSpec())
+            return edge.copy(direction = GraphDirection.UNDIRECTED_OR_RIGHT())
+        }
+
+        private fun visitEdge(ctx: GeneratedParser.EdgeAbbrevContext): GraphDirection = when {
+            ctx.TILDE() != null && ctx.ANGLE_RIGHT() != null -> GraphDirection.UNDIRECTED_OR_RIGHT()
+            ctx.TILDE() != null && ctx.ANGLE_LEFT() != null -> GraphDirection.LEFT_OR_UNDIRECTED()
+            ctx.TILDE() != null -> GraphDirection.UNDIRECTED()
+            ctx.MINUS() != null && ctx.ANGLE_LEFT() != null && ctx.ANGLE_RIGHT() != null -> GraphDirection.LEFT_OR_RIGHT()
+            ctx.MINUS() != null && ctx.ANGLE_LEFT() != null -> GraphDirection.LEFT()
+            ctx.MINUS() != null && ctx.ANGLE_RIGHT() != null -> GraphDirection.RIGHT()
+            ctx.MINUS() != null -> GraphDirection.LEFT_UNDIRECTED_OR_RIGHT()
+            else -> throw error(ctx, "Unsupported edge type")
+        }
+
+        override fun visitGraphPart(ctx: GeneratedParser.GraphPartContext): GraphPart {
+            val part = super.visitGraphPart(ctx)
+            if (part is GraphPattern) {
+                return translate(ctx) { graphMatchPattern(part) }
+            }
+            return part as GraphPart
+        }
+
+        override fun visitPatternQuantifier(ctx: GeneratedParser.PatternQuantifierContext) = translate(ctx) {
+            when {
+                ctx.quant == null -> graphQuantifier(ctx.lower.text.toLong(), ctx.upper?.text?.toLong())
+                ctx.quant.type == GeneratedParser.PLUS -> graphQuantifier(1L, null)
+                ctx.quant.type == GeneratedParser.ASTERISK -> graphQuantifier(0L, null)
+                else -> throw error(ctx, "Unsupported quantifier")
+            }
+        }
+
+        override fun visitNode(ctx: GeneratedParser.NodeContext) = translate(ctx) {
+            val variable = visitOrNull<Identifier>(ctx.symbolPrimitive())?.symbol
+            val prefilter = ctx.whereClause()?.let { visitExpr(it.expr()) }
+            val label = visitOrNull<GraphLabel>(ctx.labelSpec())
+            graphMatchNode(prefilter, variable, label)
+        }
+
+        private fun visitRestrictor(ctx: GeneratedParser.PatternRestrictorContext?): GraphRestrictor? {
+            if (ctx == null) return null
+            return when (ctx.restrictor.text.lowercase()) {
+                "trail" -> GraphRestrictor.TRAIL()
+                "acyclic" -> GraphRestrictor.ACYCLIC()
+                "simple" -> GraphRestrictor.SIMPLE()
+                else -> throw error(ctx, "Unrecognized pattern restrictor")
```

Comment:
Curious if the message ever propagates beyond here

Reply:
I tested locally with a misspelt graph restrictor (i.e. `TRAILL` rather than `TRAIL`). The message will be part of a syntax `PError` with a message captured in the "CAUSE" field -- "org.partiql.parser.PartiQLParserException: Unrecognized pattern restrictor".

> ErrorException{error=PError{code=1, severity=ERROR, kind=SYNTAX, location=null, properties={CAUSE=org.partiql.parser.PartiQLParserException: Unrecognized pattern restrictor}}}

Reply:
I haven't been following the latest `PError` developments. Is this the expected behavior?

Reply:
Yes it's ok, we explicitly state that properties are not semvered